frontend/src/index.js

```diff
@@ -124,7 +124,13 @@ async function showDirectory(dir, revision, files) {
     revision: revision || REV_LATEST,
     file_name() {
       // Build filename relative to current dir
-      return dir ? this.path.substring(dir.length + 1) : this.path;
+      if (dir) {
+        // Remove extra / only when present
+        const offset = dir.substring(dir.length - 1) === "/" ? 0 : 1;
```

Comment:
Nit: no need to get a substring here, just compare the last character: `dir[dir.length - 1] === "/"`.

Reply:
This is how i know it's time to head back home. Thanks